recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+                "apple-clang": "10",
+            },
+        }
+        return standards.get(cppstd) or {}
+
+    @property
+    def _cppstd(self):
+        return "11" if Version(self.version) <= "0.17.6" else "17"
+
+    def validate(self):
+        min_version = self._minimum_compilers_version(self._cppstd).get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._cppstd, self.settings.compiler, self.settings.compiler.version))
+
+        if self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) > "10.0" and \
+                self.settings.arch == 'x86':
+            raise ConanInvalidConfiguration("clang >= 11.0 does not support x86")
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
+        if self.options.with_openssl and self.settings.os == "Windows" and self.settings.arch == "x86":
+            raise ConanInvalidConfiguration("OpenSSL is not supported for Windows x86")
+
+    def _cmake_configure(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = self._cppstd
+            if Version(self.version) <= "0.17.6":
+                self._cmake.definitions["CAF_NO_AUTO_LIBCPP"] = True
+                self._cmake.definitions["CAF_NO_OPENSSL"] = not self.options.with_openssl
+                for define in ["CAF_NO_EXAMPLES", "CAF_NO_TOOLS", "CAF_NO_UNIT_TESTS", "CAF_NO_PYTHON"]:
+                    self._cmake.definitions[define] = "ON"
+                self._cmake.definitions["CAF_BUILD_STATIC"] = not self.options.shared
+                self._cmake.definitions["CAF_BUILD_STATIC_ONLY"] = not self.options.shared
+            else:
+                self._cmake.definitions["CAF_ENABLE_OPENSSL_MODULE"] = self.options.with_openssl
+                for define in ["CAF_ENABLE_EXAMPLES", "CAF_ENABLE_TOOLS", "CAF_ENABLE_TESTING"]:
+                    self._cmake.definitions[define] = "OFF"
+            self._cmake.definitions["CAF_LOG_LEVEL"] = self.options.log_level.value.upper()
+            if self.settings.compiler == 'clang':
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread -ldl"
+            else:
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread"
+                if self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5.0":
+                    self._cmake.definitions["CMAKE_SHARED_LINKER_FLAGS"] = "-pthread"
```

Comment:
Why is it needed? I don't see `PTHREAD_LIBRARIES` variables in upstream files.

Reply:
Yep, you're right, as of 0.17.5 it's no longer used.